### PR TITLE
ver: bump version to 0.4.2

### LINKED ISSUE #38: [BUG] build wheels for `manylinux`
## Describe the bug

Unsupported platform tag for wheels built in GitHub Action.

https://github.com/metaopt/TorchOpt/runs/7430789533?check_suite_focus=true

```
ERROR    HTTPError: 400 Bad Request from https://test.pypi.org/legacy/
         Binary wheel 'torchopt-0.4.2a0-cp310-cp310-linux_x86_64.whl' has an    
         unsupported platform tag 'linux_x86_64'.                               
Error: Process completed with exit code 1.
```

## To Reproduce

```console
$ python3 -m pip install --upgrade pip setuptools build auditwheel

$ python3 -m build

$ auditwheel show dist/torchopt-0.4.2-cp38-cp38-linux_x86_64.whl

torchopt-0.4.2-cp38-cp38-linux_x86_64.whl is consistent with the
following platform tag: "linux_x86_64".

The wheel references external versioned symbols in these
system-provided shared libraries: libgomp.so.1 with versions
{'OMP_1.0', 'GOMP_4.0'}, librt.so.1 with versions {'GLIBC_2.2.5'},
libgcc_s.so.1 with versions {'GCC_3.0', 'GCC_3.3.1'}, libm.so.6 with
versions {'GLIBC_2.2.5'}, libc.so.6 with versions {'GLIBC_2.4',
'GLIBC_2.2.5', 'GLIBC_2.3', 'GLIBC_2.3.3', 'GLIBC_2.6'},
libstdc++.so.6 with versions {'CXXABI_1.3.5', 'CXXABI_1.3',
'GLIBCXX_3.4.9', 'CXXABI_1.3.3', 'GLIBCXX_3.4.18', 'GLIBCXX_3.4',
'CXXABI_1.3.2'}, libdl.so.2 with versions {'GLIBC_2.2.5'},
libpthread.so.0 with versions {'GLIBC_2.3.2', 'GLIBC_2.3.4',
'GLIBC_2.2.5'}

This constrains the platform tag to "manylinux_2_17_x86_64". In order
to achieve a more compatible tag, you would need to recompile a new
wheel from source on a system with earlier versions of these
libraries, such as a recent manylinux **image.**
```

References:

- https://github.com/pypa/manylinux
- https://github.com/pypa/python-manylinux-demo
- https://github.com/pypa/auditwheel

## Checklist

- [X] I have checked that there is no similar issue in the repo (**required**)
- [X] I have read the [documentation](https://torchopt.readthedocs.io/) (**required**)
- [X] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by XuehaiPan**
## Description

Describe your changes in detail.

## Motivation and Context

Bump version and add changelog.

Closes #38

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [X] I have read the [CONTRIBUTION](https://torchopt.readthedocs.io/en/latest/developer/contributing.html) guide (**required**)
- [X] My change requires a change to the documentation.
- [X] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [X] I have updated the documentation accordingly.
- [X] I have reformatted the code using `make format` (**required**)
- [X] I have checked the code using `make lint` (**required**)
- [X] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,14 @@
   <img src="image/logo-large.png" width="75%" />
 </div>
 
+![Python 3.7+](https://img.shields.io/badge/Python-3.5%2B-brightgreen.svg)
```
Comment (Benjamin-eecs) [inline README.md:8]: add unittest passing badge, issue and fork badge, coverage badge later maybe

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+[build-system]
+requires = ["setuptools", "torch == 1.12", "numpy", "pybind11"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "torchopt"
+description = "A Jax-style optimizer for PyTorch."
```
Comment (Benjamin-eecs) [inline pyproject.toml:7]: lack long description from readme, it's a must

Comment (XuehaiPan) [reply pyproject.toml:7]: The long description is specified by `readme = "README.md"` below.

**Review by Benjamin-eecs — APPROVED** [review]
LGTM!

**Pull request merged**